documentation/proposals/Proposal - Generic Math.md

```diff
@@ -0,0 +1,366 @@
+- A Clamp function which takes a Max vector and a Min vector and returns a vector which has its components bounded between the Min and Max vectors.
+  - A Static implementation of this function **must** be available as well.
+- A Clamp function which takes a Max scalar and a Min scalar, both which match the generic type, and returns a vector which has its components bounded between the Min and Max scalars.
+  - A Static implementation of this function **must** be available as well.
+- An Abs function which returns a vector where each component is the absolute value of the original
+  - A Static implementation of this function **must** be available as well.
+- CopyTo functions which copy to an array, with or without a starting index
+- Explicit cast and checked cast operators to all standard variants of the F and I vector types of the same dimensionality
+- Explicit cast and checked cast to and from matching System.Numerics vector type
+- Explicit cast cast to lower dimensional matching vector with matching generic type
+- A CopySign function which takes a vector and copies the signs component-wise to our vector
+  - A Static implementation of this function **must** be available but it should return a new vector without affecting the original
+- A Copy sign function which takes a scalar which matches the generic type, and copies the scalars sign onto each component of the vector
+  - A Static implementation of this function **must** be available but it should return a new vector without affecting the original
+- A Sign function which returns a vector where each component is only the sign segment of the original vector
+  - A Static implementation of this function **must** be available
+- Static Unit Vectors for each component
+- A Static Zero Vector with zero for all components
+- A Static One Vector with one for all components
+- A static AllBitsSet Vector with all bits set for all components
+- Define static CreateChecked, CreateSaturating, and CreateTruncating which converts other vector types to this type
+  - Try variants of these methods should also be defined which out the resulting vector and return a bool representing success or failure of the operation.
+- Define Transform functions which take a Matrix of higher dimensionality (Vector 2 can use Matrix2xn, Matrix3xn, and matrix4xn) and return a vector containing the output (type should match the outer type e.g. Vector2.Transform(Matrix4x4) returns Vector2)
+  - A Static implementation of these functions **must** be available
+  - Define matching `*` operators which function the same
+- Define TransformNormal functions which take a Matrix of higher dimensionality (Vector 2 can use Matrix2xn, Matrix3xn, and matrix4xn) and return a vector containing the output (type should match the outer type e.g. Vector2.Transform(Matrix4x4) returns Vector2)
+  - A Static implementation of these functions **must** be available
+
+For I types, the following requirements **must** also be fulfilled:
```

Comment:
following additional requirements and remove the also maybe? would read better to me that way
```suggestion
For I types, the following additional requirements **must** be fulfilled:
```